Networking

JockeBTH opened the pull request:
Snapshots are now sent on field level and not component level.
Packets now dynamically allocates more memory when they need it.
Increased local receive buffer for Client and Server() and removed the magic number that was used
In Server.h and Client.h. INPUTSIZE is now used instead.
(char readBuffer[INPUTSIZE] = { 0 }).
Packets start size has now been increased to 512 bytes.
Changed ComponentInfo::FieldsInOrder to store strings instead of pointers.
Removed HasEntity() and are now using ValidEntity() instead.

Stiffly committed:
Removed some comments. Basically did nothing

JockeBTH committed:
Removed unnecessary code.

JockeBTH committed:
WIP Model component not working.

JockeBTH committed:
Packets now dynamically allocates more memory when they need it.

Increased local receive buffer for Client and Server() and removed the magic number that was used
In Server.h and Client.h
(char readBuffer[INPUTSIZE] = { 0 }).
Packets start size has now been increased to 512 bytes.
Changed ComponentInfo::FieldsInOrder to store strings instead of pointers.
Removed HasEntity() and are now using ValidEntity() instead.

JockeBTH committed:
Merge remote-tracking branch 'origin/master' into Networking

JockeBTH committed:
Removed debug code from Game.cpp

wima12 commented on src/Engine/Network/Client.cpp:
```diff
@@ -46,16 +40,15 @@ void Client::Update()
-        m_EventBroker->Unsubscribe(m_EInputCommand);
-    }
+    disconnect();
+    m_ThreadIsRunning = false;
```
Remove threadrunning, and Close() methods in Network, Client, and Server.

JockeBTH committed:
Removed Close method from Client, Server and Network.

Removed legacy variable m_ThreadIsRunning in Client.h and Server.h.
Removed m_EventBroker->Unsubscribe(m_EInputCommand) in Client destructor.
